src/linux_serial_ccsds/linux_serial_ccsds.cpp

```diff
@@ -0,0 +1,231 @@
+{
+    if(device->use_paritybit) {
+        *cflags |= PARENB;
+        switch(device->parity) {
+            case odd:
+                *cflags |= PARODD;
+                break;
+            case even:
+                *cflags &= ~PARODD;
+                break;
+            default:
+                *cflags &= ~PARENB;
+                std::cerr << "Not supported parity type deaulting to no parity";
+        }
+    } else {
+        *cflags &= ~PARENB;
+    }
+}
+
+void
+linux_serial_ccsds_private_data::driver_init(const SystemBus bus_id,
+                                             const SystemDevice device_id,
+                                             const Serial_CCSDS_Linux_Conf_T* const device_configuration,
+                                             const Serial_CCSDS_Linux_Conf_T* const remote_device_configuration)
+{
+    m_serial_device_bus_id = bus_id;
+    m_serial_device_id = device_id;
+    m_serial_device_configuration = device_configuration;
+    m_serial_remote_device_configuration = remote_device_configuration;
+    /// Open UART device
+    /**
+     * Access mode      O_RDWR - read write access mode
+     * Blocking mode    O_NDELAY - non blocking mode
+     * File type        O_NOCTTY - pathname will refer to tty
+     */
+    serialFd = open(device_configuration->devname, O_RDWR | O_NOCTTY);
+    if(serialFd == -1) {
+        std::cerr << "Error while opening a file \n\r";
+        exit(EXIT_FAILURE);
+    }
+
+    /// Configure UART
+    struct termios options;
+
+    int cflags = 0;
+
+    driver_init_baudrate(device_configuration, &cflags);
+    driver_init_character_size(device_configuration, &cflags);
+    driver_init_parity(device_configuration, &cflags);
+
+    tcgetattr(serialFd, &options);
+    options.c_cflag = cflags | CLOCAL | CREAD;
+    options.c_iflag = IGNPAR;
+    options.c_oflag = 0;
+    options.c_lflag = 0;
+    tcflush(serialFd, TCIFLUSH);
+    tcsetattr(serialFd, TCSANOW, &options);
+
+    m_thread.start(&taste::LinuxSerialCcsdsPoll, this);
+}
+
+void
+linux_serial_ccsds_private_data::driver_poll()
+{
+    ssize_t length{ 0 };
+    Escaper_start_decoder(&escaper);
+    while(1) {
+        if(serialFd != -1) {
+            length = read(serialFd, m_recv_buffer, DRIVER_RECV_BUFFER_SIZE);
+            if(length > 0) {
+                Escaper_decode_packet(&escaper, m_recv_buffer, length, Broker_receive_packet);
+            } else {
+                std::cerr << "Error while polling. Cannot read.\n\r";
+                exit(EXIT_FAILURE);
+            }
+        } else {
+            std::cerr << "Error while polling. Wrong file descriptor\n\r";
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
+void
+linux_serial_ccsds_private_data::driver_send(const uint8_t* const data, const size_t length)
+{
+    if(serialFd != -1) {
+        Escaper_start_encoder(&escaper);
+        size_t index = 0;
+        size_t packetLength = 0;
+
+        while(!escaper.m_encode_finished) {
```

Comment:
The function `Escaper_encode_finished()`? Or `index < length`. This looks like "accessing private data".